mantidproject/mslice
# Waterfall plots

This PR adds a waterfall plot option to `Cut` figures.

**To test:**

1. Load a dataset and make several cuts - either using the `width` option or with `Plot Over`.
2. Click on the `Waterfall` button on the plot toolbar. You should see two line-edit boxes appear to the left of the waterfall button with zeros in them - this is the *x* and *y* offsets, in data coordinates
3. Put in some numbers in the edit boxes and check that the plotted curves are shifted by the appropriate amounts.
4. Click on the `Waterfall` button again. Check that the curves are back to their original positions (e.g. the plot is no longer a waterfall) and that the *x* and *y* offset line edit boxes have disappeared.
5. Plot another cut on the same figure (using `Plot Over`), and click on `Waterfall` again - you should see that not only the original cuts have been offset, but also the new cut. 
6. Plot yet another cut on the same figure and check that it too is appropriately displaced.
7. In the commandline (IPython console) run `gca().set_waterfall(False)` and check that the waterfall effect is turned off.
8. In the commandline run `gca().set_waterfall(True)` to turn it on again, or specify a different offset using e.g. `gca().set_waterfall(True, x_offset=1, y_offset=2)` or similar.
9. In the cut plot window, enable the waterfall plot again and then generate a script file of the plot. Look in the script file and check that there is a call to `set_waterfall` with the correct offsets. Also check that the generated file runs ok and produces the correct waterfall plot.
10. Go back to the main GUI and plot a slice. Click `Interactive Cut` on this slice and select an area of the Slice plot. Check that the original cut window has now changed to an interactive cut window (and that the `Waterfall` button is hidden). Click on `Interactive Cut` on the slice window to disable the interactive cut and check that the `Waterfall` button is visible again with its previous setting (e.g. if it was previously unchecked that it is still unchecked and if it was previously checked that the *x* and *y* offset boxes are now visible with their previous values).


Fixes #433
